OpenRA.Mods.Common/Activities/AttackActivity.cs

```diff
@@ -0,0 +1,199 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Activities
+{
+	class AttackActivity : Activity, IActivityNotifyStanceChanged
```

Comment:
Any chance you could split the `Attack` -> `AttackActivity` rename/replacement to a separate commit (maybe rename the old AttackActivity to OldAttackActivity temporarily for the transition)?

My concern here is that the way the commit is now, comparing the diff between the old Attack.cs and this is tricky, and risks silent rebase conflicts with bleed.

Reply:
It's actually the other way around. `Attack` is not renamed to `AttackActivity`; It is the `AttackActivity` from `AttackFollow` that has been split off into it's own file with some small adjustments. the old `Attack` is completely defunct.

Reply:
If it helps with reviewing, the changes compared to the old `AttackActivity` are:
- The addition of the block beneath "Turn to face the target if required"
- the addition of the `canMove` bool and associated changes.

Anything else could be a potential rebase conflict, although after half a year I am a bit fuzzy on the details.